client/src/charts/nivo/nivo_line.js

```diff
@@ -0,0 +1,222 @@
+import './NivoCharts.scss';
+import { ResponsiveLine } from '@nivo/line';
+import classNames from 'classnames';
+import {
+  graph_text_maker,
+  InteractiveGraph,
+  general_default_props,
+  default_tooltip,
+  fixed_symbol_shape,
+  get_formatter,
+} from './nivo_shared.js';
+import {
+  DisplayTable,
+} from '../../components/index.js';
+import { IconZoomIn, IconZoomOut } from '../../icons/icons.js';
+
+const get_scale_bounds = (stacked, raw_data, zoomed) => {
+  const min = _.min(raw_data);
+  const max = _.max(raw_data);
+  const scaled_min = min < 0 ? min * 1.05 : min * 0.95;
+  const scaled_max = max < 0 ? max * 0.95 : max * 1.05;
+  if(stacked){
+    return {
+      min: min < 0 ? scaled_min : 0,
+      max: 'auto',
+    };
+  }
+  return {
+    min: zoomed || min < 0 ? scaled_min : 0,
+    max: !zoomed && max < 0 ? 0 : scaled_max,
+  };
+};
+
+
+export class NivoResponsiveLine extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      y_scale_zoomed: false,
+    };
+  }
+  render(){
+    const {
+      data,
+      raw_data,
+      margin,
+      legends,
+      graph_height,
+      bttm_axis,
+      left_axis,
+      remove_bottom_axis,
+      remove_left_axis,
+      disable_yaxis_zoom,
+      yScale,
+      enableDotLabel,
+      is_money,
+      text_formatter,
+      theme,
+      colors,
+      colorBy,
+      tooltip,
+      max, // only used for welcome mat
+      min, // only used for welcome mat
+      enableArea,
+      enableGridX,
+      enableGridY,
+      stacked,
+      markers,
+      layers,
+      disable_table_view,
+      table_name,
+      table_first_column_name,
+      isInteractive,
+      motionDamping,
+      motionStiffness,
+    } = this.props;
+
+    const {
+      y_scale_zoomed,
+    } = this.state;
+
+    const IE_fixed_legends = legends ? (
+      _.map(legends,
+        legend => _.chain(legend)
+          .clone()
+          .assign({symbolShape: fixed_symbol_shape})
+          .value() )
+      ) :
+      undefined;
```

Comment:
Oh, here too, guess this should be pulled back to `../nivo_shared.js`

Reply:
Good catch!